### PR TITLE
String taint tracking - repeat method

### PR DESCRIPTION
**Pull request by jandro996**
# What Does This Do
Adds all the instrumentation needed to perform taint tacking in repeat operation for String objects

# Motivation
IAST requires to track all modifications that happen to strings in the code, this PR implements the repeat operation for String class

# Additional Notes
The  String repeat method is available since JAVA 11

### CODE REVIEW

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/IastModuleImpl.java [hunk 1]
```diff
@@ -191,6 +193,27 @@ public void onStringBuilderToString(
     taintedObjects.taint(result, to.getRanges());
   }
 
+  @Override
+  public void onStringRepeat(String self, int count, String result) {
+    if (!canBeTainted(self) || !canBeTainted(result) || self == result) {
```
Comment (manuel-alvarez-alvarez) [inline dd-java-agent/agent-iast/src/main/java/com/datadog/iast/IastModuleImpl.java:198]: You can replace `self == result` by 'count <= 1'

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/propagation/StringModuleImpl.java [hunk 2]
```diff
@@ -233,6 +233,28 @@ public void onStringJoin(
...
+  public void onStringRepeat(String self, int count, String result) {
+    if (!canBeTainted(self) || !canBeTainted(result) || self == result) {
+      return;
+    }
+    final IastRequestContext ctx = IastRequestContext.get();
+    if (ctx == null) {
+      return;
+    }
+    final TaintedObjects taintedObjects = ctx.getTaintedObjects();
+    final Range[] selfRanges = getRanges(taintedObjects.get(self));
```
Comment (smola) [inline dd-java-agent/agent-iast/src/main/java/com/datadog/iast/propagation/StringModuleImpl.java:247]: Is `getRanges` guaranteed to return not-null?

File: dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/propagation/StringModuleTest.groovy [hunk 3]
```diff
@@ -655,6 +655,77 @@ class StringModuleTest extends IastModuleImplTestBase {
     ]                                                                                                                    | "stringParam1,stringParam2,stringParam3:==>taintedString<==, ==>taintedString<==, ==>taintedString<=="
   }
 
+  void 'onStringRepeat that can not be tainted (#self, #count)'(final String self, final int count, final String expected) {
+    when:
+    module.onStringRepeat(self, count, expected)
+    then:
```
Comment (smola) [inline dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/propagation/StringModuleTest.groovy:661]: Space out blocks in tests, please.

**Review by smola — CHANGES_REQUESTED** [review]

File: dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/propagation/StringModuleTest.groovy (hunk 3 above)
Comment (jandro996) [reply dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/propagation/StringModuleTest.groovy:662]: 🤦

File: dd-java-agent/agent-iast/src/main/java/com/datadog/iast/propagation/StringModuleImpl.java (hunk 2 above)
Comment (jandro996) [reply dd-java-agent/agent-iast/src/main/java/com/datadog/iast/propagation/StringModuleImpl.java:247]: Yes!

```
private static Range[] getRanges(final TaintedObject taintedObject) {
    return taintedObject == null ? Ranges.EMPTY : taintedObject.getRanges();
  }
```
and

```
public TaintedObject(
      final @Nonnull Object obj,
      final @Nonnull Range[] ranges,
      final @Nullable ReferenceQueue<Object> queue) {
    super(obj, queue);
    this.positiveHashCode = System.identityHashCode(obj) & POSITIVE_MASK;
    this.ranges = ranges;
  }
```

File: dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/propagation/StringModuleTest.groovy [hunk 4]
```diff
@@ -759,18 +838,18 @@ class StringModuleTest extends IastModuleImplTestBase {
...
+    "i̇̀==>def<==b"                       | "ÌD==>EFB<=="              | "lt"   | 7          | 6            | [[3, 3]]
+    "i̇̀==>def<==b"                       | "İ̀==>DEF<==B"             | "en"   | 7          | 7            | [[3, 3]]
+    "i̇̀==>def<==b==>def<=="              | "İ̀==>DEF<==B==>DEF<=="    | "en"   | 10         | 10           | [[3, 3], [7, 3]]
     "\u00cc==>def<==b"                    | "\u00cc==>DEF<==B"          | "lt"   | 5          | 5            | [[1, 3]]
-    "i̇̀i̇̀==>fff<==f123b"                    | "ÌÌFF==>FF1<==23B"          | "lt"   | 14         | 12           | [[6, 3]]
-    "i̇̀i̇̀i̇̀i̇̀EEEE==>fff<=="                   | "ÌÌÌÌEEEEFFF"               | "lt"   | 19         | 15           | []
-    "i̇̀i̇̀i̇̀i̇̀EEEE==>fff<==H==>GGG<=="         | "ÌÌÌÌEEEEFFFH==>GGG<=="     | "lt"   | 23         | 19           | [[16, 3]]
-    "i̇̀i̇̀i̇̀EEEE==>fffgggg<=="                | "ÌÌÌEEEEFFF==>GGGG<=="      | "lt"   | 20         | 17           | [[13, 4]]
+    "i̇̀i̇̀==>fff<==f123b"                | "ÌÌFF==>FF1<==23B"        | "lt"   | 14         | 12           | [[6, 3]]
```
Comment (smola) [inline dd-java-agent/agent-iast/src/test/groovy/com/datadog/iast/propagation/StringModuleTest.groovy:849]: @jandro996 Please, reformat this.

**Review by smola — APPROVED** [review]

**Pull request merged**